subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -102,15 +102,46 @@ static inline bool isr_rx_ci_adva_check(uint8_t tx_addr, uint8_t *addr,
 #endif
 
 #define PDU_MEM_SIZE       PDU_ADV_MEM_SIZE
-#define PDU_MEM_COUNT_MIN  (BT_CTLR_ADV_SET + \
-			    (BT_CTLR_ADV_SET * PAYLOAD_FRAG_COUNT) + \
-			    (BT_CTLR_ADV_AUX_SET * PAYLOAD_FRAG_COUNT) + \
-			    (BT_CTLR_ADV_SYNC_SET * PAYLOAD_FRAG_COUNT))
-#define PDU_MEM_FIFO_COUNT ((BT_CTLR_ADV_SET * PAYLOAD_FRAG_COUNT * 2) + \
-			    (CONFIG_BT_CTLR_ADV_DATA_BUF_MAX * \
+
+/* AD data and Scan Response Data need 2 PDU buffers each in the double buffer
+ * implementation. Allocate 3 PDU buffers plus CONFIG_BT_CTLR_ADV_DATA_BUF_MAX
+ * defined 1 PDU buffer as the minimum number of buffers that meet the legacy
+ * advertising needs. Add 1 each for Extended and Periodic Advertising, needed
+ * extra for double buffers for these is kept as configurable, by increasing
+ * CONFIG_BT_CTLR_ADV_DATA_BUF_MAX.
+ */
+#define PDU_MEM_COUNT_MIN  ((BT_CTLR_ADV_SET * 3) + \
+			    ((BT_CTLR_ADV_AUX_SET + \
+			      BT_CTLR_ADV_SYNC_SET) * \
+			     PAYLOAD_FRAG_COUNT))
+
+/* Maximum advertising PDU buffers to allocate, which is the sum of minimum
+ * plus configured additional count in CONFIG_BT_CTLR_ADV_DATA_BUF_MAX.
+ */
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+#define PDU_MEM_COUNT_MAX ((PDU_MEM_COUNT_MIN) + \
+			   ((1 + CONFIG_BT_CTLR_ADV_DATA_BUF_MAX) * \
```

Comment:
Why do we only need 1 additional buffer for periodic advertisement? Wouldn't we need 1 buffer per advertising set?

Reply:
When periodic advertising is supported, then the minimum periodic advertising instance count is 1 with a dependency of 1 extended advertising instance. `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` is 1 from Kconfing, hence 1 additional is added for a minimum periodic advertising instance. If more periodic advertising is enabled (count increased) then `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` should be increased based on the applications behavior of updates to PDU data.

If the Extended advertising data of all the sets is updated at the same time then both the elements of the double buffers of these sets would need allocation compared to application use cases where AD data is seldom updated. Hence, the required spare PDU buffers be configured using `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` by applications behavior.